colectica_api/api.py

```diff
@@ -764,6 +764,77 @@ def get_a_set_lasted(self, AgencyId, Identifier):
             if response.json() != []:
                 return response.json()
 
+    def update_state(
+        self,
+        items,
+        *,
+        State=False,
+        ApplyToAllVersions=True
+    ):
+        """Updates the Deprecated State of a set of items.
+        
+        Args:
+            items: For example, ``[
+                {
+                    "agencyId": "uk.closer",
+                    "identifier": "9da68988-f9e4-48b3-996b-fbcbd3c5d1f6",
+                    "version": 2
+                }
+            ]``. All three of agencyId, identifier and version must be
+            specified in order for the deprecation status of an item to be
+            updated.
+
+        Keyword Args:
+            State: (boolean/None): if omitted, item's deprecated state is
+                set to 'false'.
+            ApplyToAllVersions: if omitted, all versions of item have their
+                deprecation status updated.    
+        
+        Returns:
+            HTTP status code indicating success or failure of operation, e.g.
+            success=200
+    
+        This uses the ``/api/v1/item/_updateState`` API call.
+
+        Documented here: https://docs.colectica.com/portal/technical/api/v1/#tag/Item/paths/~1api~1v1~1item~1_updateState/post
+        """
+
+        items_lower_case_fields = [{key.lower(): value for key, value in dictionary.items()} for dictionary in items]
+        itemsWithNoAgencyId=[i for i, e in enumerate(['agencyid' in x for x in items_lower_case_fields]) if e == False]
+        
+        itemsWithNoIdentifier=[i for i, e in enumerate(['identifier' in x for x in items_lower_case_fields]) if e == False]
+        
+        itemsWithNoVersion=[i for i, e in enumerate(['version' in x for x in items_lower_case_fields]) if e == False]
+        
+        if (len(itemsWithNoAgencyId)>0):
+            print("\nThe following item(s) did not have an agency id specified: ")
+            print([items[i] for i in itemsWithNoAgencyId])
+
+        if (len(itemsWithNoIdentifier)>0):
+            print("\nThe following item(s) did not have an identifier specified: ")
+            print([items[i] for i in itemsWithNoIdentifier])
+                
+        if (len(itemsWithNoVersion)>0):
+            print("\nThe following item(s) did not have a version specified: ")
+            print([items[i] for i in itemsWithNoVersion])
+            
+        if (len(itemsWithNoAgencyId)>0 or len(itemsWithNoIdentifier)>0 or len(itemsWithNoVersion)>0):
+            print("\nPlease ensure that all elements in the items array contain agencyId, identifier, and version fields.")
+            return    
```

Comment:
Do you think this should raise an exception instead?   For example:
```
raise KeyError("all elements in the items array contain agencyId, identifier, and version fields")
```
or ValueError?

Reply:
Good spot; I'm not used to programming in Python, so I didn't think to raise an exception here. I've pushed up a commit which raises a KeyError if the input isn't correct.